bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitUiUpdatePublisher.java

```diff
@@ -44,9 +51,33 @@ public void changed(ProgressEvent event) {
     }
 
     public void completed(ProgressEvent event) {
-        PitUiUpdate update = new PitUiUpdate.Builder().withHtml(
-                browser.getText()).build();
+        PitUiUpdate update = new PitUiUpdate.Builder().withHtml(getHtml()).build();
         handler.execute(Platform.getExtensionRegistry(), update);
     }
 
+    /**
+     * Get the index.html of the results, even if the browser shows a different file
+     * of the results
+     * @return html text of the index.html as a String
+     */
+    private String getHtml() {
+        // assume url is path to index.html and top folder should represent date and
+        // time
+        // remove os differences in path
+        final Pattern pattern;
+        if (WINDOWS) {
```

Comment:
This will be impossible to cover since we do code coverage only in Linux.
Moreover, I'd avoid solutions that are based on the OS.
That's also why I suggested to focus on the browser navigation buttons and postpone this other feature to the future (though, as I said, I doubt it can be implemented safely since it highly relies on the format of the HTML that might change in the future versions of PIT, making it a problem to maintain).

In any case, if you want to implement such a mechanism, I'd strongly suggest you factor out this code into a separate class that does not need the UI and can be tested in isolation, e.g., in `runner.tests`. For example, you can put into the `runner.tests` a directory with some contents and run tests against such contents.